> > Workflow Check failed [master-build](https://github.com/asterisk/asterisk/actions/runs/25756666014/job/75647196307): FAILED: Build
> 
> Yes. It literally fails to build.

With -Wall -Werror sure.  So this correct warning once I fix the boolean situation (which for some reason I missed):

```
res_pjsip/pjsip_configuration.c: In function ‘dtlsverify_to_str’:
res_pjsip/pjsip_configuration.c:1034:9: warning: case value ‘3’ not in enumerated type ‘enum ast_rtp_dtls_verify’ [-Wswitch]
 1034 |         case AST_RTP_DTLS_VERIFY_FINGERPRINT|AST_RTP_DTLS_VERIFY_CERTIFICATE:
      |         ^~~~
```

Two fixes:

1.  Add a BOTH value to the enum.  Which frankly in my opinion is not the right way since this enum (by virtue of using bit-shifts in the enum definition) clearly represents a bitmask.
2.  Explicit cast to int so the enum value check isn't performed.  This is also nasty but the less nasty of the two options, and thus the route I went with.

Seems I screwed up copying back from the test host to where I run my git push from during previous push.  Re-verified. And will push now.

--- a/res/res_pjsip/pjsip_configuration.c
+++ b/res/res_pjsip/pjsip_configuration.c
@@ -1021,7 +1021,19 @@ static int dtls_handler(const struct aco_option *opt,
 static int dtlsverify_to_str(const void *obj, const intptr_t *args, char **buf)
 {
 	const struct ast_sip_endpoint *endpoint = obj;
-	*buf = ast_strdup(AST_YESNO(endpoint->media.rtp.dtls_cfg.verify));
+	if (endpoint->media.rtp.dtls_cfg.verify == AST_RTP_DTLS_VERIFY_NONE) {
+		*buf = ast_strdup("No");
+	} else if (endpoint->media.rtp.dtls_cfg.verify == AST_RTP_DTLS_VERIFY_FINGERPRINT) {
+		*buf = ast_strdup("Fingerprint");
+	} else if (endpoint->media.rtp.dtls_cfg.verify == AST_RTP_DTLS_VERIFY_CERTIFICATE) {
+		*buf = ast_strdup("Certificate");
+	} else if (endpoint->media.rtp.dtls_cfg.verify == (AST_RTP_DTLS_VERIFY_FINGERPRINT|AST_RTP_DTLS_VERIFY_CERTIFICATE)) {
+		*buf = ast_strdup("Yes");
+	} else {
+		*buf = NULL;
+		ast_assert(0);
+		return 1;
+	}
 	return 0;
 }
 

